core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -7236,6 +7236,30 @@ public SqlNode go(SqlNode root) {
           () -> this + " returned null for " + root);
     }
 
+    /** True if the exception ex indicates that name lookup has failed. */
+    public static boolean isNotFoundException(Exception ex) {
+      if (!(ex instanceof CalciteContextException)) {
+        return false;
+      }
+      String message = ex.getMessage();
+      if (message != null && !message.contains("not found")) {
```

Comment:
A little hard code, here seems can use StringUtils.contains(message, "not found") to instead.

Reply:
This judgment is somewhat repetitive, and the effect of StringUtils.contains should be the same as these judgments. 
this is StringUtils.contains related codes:
```
    public static boolean contains(String str, String searchStr) {
        if (str == null || searchStr == null) {
            return false;
        }
        return str.indexOf(searchStr) >= 0;
    }

```
It's just a suggestion, no problem.

Reply:
turns out that StringUtils.contains does not work, since it is not annotated with `@Nullable`, and thus is causes the CheckedFramework to reject the program.

Reply:
OK，keep it as current version is better.